CarRental.Api.Host/Controllers/ReportsController.cs

```diff
@@ -0,0 +1,169 @@
+    [ProducesResponseType(204)]
+    [ProducesResponseType(500)]
+    public ActionResult<IEnumerable<CurrentlyRentedCarDto>> GetCarsCurrentlyRented()
+    {
+        logger.LogInformation("{method} method of {controller} is called",
+            nameof(GetCarsCurrentlyRented), GetType().Name);
+
+        try
+        {
+            var rentedCars = reports.GetCarsCurrentlyRented();
+
+            logger.LogInformation("{method} method of {controller} executed successfully",
+                nameof(GetCarsCurrentlyRented), GetType().Name);
+
+            return rentedCars.Any() ? Ok(rentedCars) : NoContent();
+        }
+        catch (Exception ex)
+        {
+            logger.LogError("An exception happened during {method} method of {controller}: {@exception}",
+                nameof(GetCarsCurrentlyRented), GetType().Name, ex);
+
+            return StatusCode(500, $"{ex.Message}\n\r{ex.InnerException?.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Get top 5 most frequently rented cars
+    /// </summary>
+    /// <returns>Top 5 cars with rental counts</returns>
+    [HttpGet("top-5-rented-cars")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(500)]
+    public ActionResult<IEnumerable<CarWithRentalCountDto>> GetTop5MostFrequentlyRentedCars()
+    {
+        logger.LogInformation("{method} method of {controller} is called",
+            nameof(GetTop5MostFrequentlyRentedCars), GetType().Name);
+
+        try
+        {
+            var topCars = reports.GetTop5MostFrequentlyRentedCars();
+
+            logger.LogInformation("{method} method of {controller} executed successfully",
+                nameof(GetTop5MostFrequentlyRentedCars), GetType().Name);
+
+            return topCars.Any() ? Ok(topCars) : NoContent();
+        }
+        catch (Exception ex)
+        {
+            logger.LogError("An exception happened during {method} method of {controller}: {@exception}",
+                nameof(GetTop5MostFrequentlyRentedCars), GetType().Name, ex);
+
+            return StatusCode(500, $"{ex.Message}\n\r{ex.InnerException?.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Get rental count for each car
+    /// </summary>
+    /// <returns>Dictionary with car ID and rental count</returns>
+    [HttpGet("rental-count-per-car")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(500)]
+    public ActionResult<Dictionary<int, int>> GetRentalCountPerCar()
+    {
+        logger.LogInformation("{method} method of {controller} is called",
+            nameof(GetRentalCountPerCar), GetType().Name);
+
+        try
+        {
+            var rentalCounts = reports.GetRentalCountPerCar();
+
+            logger.LogInformation("{method} method of {controller} executed successfully",
+                nameof(GetRentalCountPerCar), GetType().Name);
+
+            return rentalCounts.Any() ? Ok(rentalCounts) : NoContent();
```

Comment:
```suggestion
            return Ok(rentalCounts);
```
Пусть если в коллекции нет элементов возвращается коллекция без элементов.